crypto/evp/keymgmt_lib.c

```diff
@@ -459,6 +459,12 @@ int evp_keymgmt_util_copy(EVP_PKEY *to, EVP_PKEY *from, int selection)
             evp_keymgmt_freedata(to_keymgmt, alloc_keydata);
             return 0;
         }
+    } else if (to_keymgmt == from->keymgmt && to_keymgmt->dup != NULL
+               && to_keydata == NULL
```

Comment:
shouldnt the to_keydata check be inside the {} block ?

Reply:
Actually I think we can use dup even with non-NULL to_keydata, I'll implement that.

Reply:
In the end I decided it would be too risky to drop the original data so no, we need to keep it here and use the import/export to override the original data if we have some. So the to_keydata == NULL has to stay here.